### ISSUE TITLE
1000ms - 5000ms lag spikes with crash

### ISSUE THREAD

**Issue by LunarsWorldLLC** [PaperSpigot/Paper#9372]
### Stack trace

```
[02:15:05] [Paper Watchdog Thread/ERROR]: --- DO NOT REPORT THIS TO PAPER - THIS IS NOT A BUG OR A CRASH  - git-Paper-36 (MC: 1.20.1) ---
[02:15:05] [Paper Watchdog Thread/ERROR]: The server has not responded for 10 seconds! Creating thread dump
[02:15:05] [Paper Watchdog Thread/ERROR]: ------------------------------
[02:15:05] [Paper Watchdog Thread/ERROR]: Server thread dump (Look for plugins here before reporting to Paper!):
[02:15:05] [Paper Watchdog Thread/ERROR]: [ChunkTaskScheduler] Chunk wait task info below: 
[02:15:05] [Paper Watchdog Thread/ERROR]: [ChunkTaskScheduler] Chunk wait: [( -914,417) in 'resourceworld']
[02:15:05] [Paper Watchdog Thread/ERROR]: [ChunkTaskScheduler] Chunk holder: NewChunkHolder{world=resourceworld, chunkX=-914, chunkZ=417, entityChunkFromDisk=false, lastChunkCompletion={chunk_class=net.minecraft.world.level.chunk.ProtoChunk,status=minecraft:structure_starts}, currentGenStatus=minecraft:structure_starts, requestedGenStatus=minecraft:full, generationTask=null, generationTaskStatus=null, priority=BLOCKING, priorityLocked=false, neighbourRequestedPriority=IDLE, effective_priority=BLOCKING, oldTicketLevel=33, currentTicketLevel=33, totalNeighboursUsingThisChunk=0, fullNeighbourChunksLoadedBitset=0, chunkStatusRaw=0, currentChunkStatus=INACCESSIBLE, pendingChunkStatus=INACCESSIBLE, is_unload_safe=ticket_level, killed=false}
[02:15:05] [Paper Watchdog Thread/ERROR]: Ticking entity: minecraft:player, entity class: net.minecraft.server.level.EntityPlayer
[02:15:05] [Paper Watchdog Thread/ERROR]: Entity status: removed: false, valid: true, alive: true, is passenger: false
[02:15:05] [Paper Watchdog Thread/ERROR]: Entity UUID: 635213e4-d22c-4ab8-99f5-f0621d16bd23
[02:15:05] [Paper Watchdog Thread/ERROR]: Position: world: 'resourceworld' at location (43.5, 87.0, -2674.5)
[02:15:05] [Paper Watchdog Thread/ERROR]: Velocity: (0.0, 0.0, 0.0) (in blocks per tick)
[02:15:05] [Paper Watchdog Thread/ERROR]: Entity AABB: AABB[43.19999998807907, 87.0, -2674.800000011921] -> [43.80000001192093, 88.79999995231628, -2674.199999988079]
[02:15:05] [Paper Watchdog Thread/ERROR]: Total packets processed on the main thread for all players: 60355629
[02:15:05] [Paper Watchdog Thread/ERROR]: ------------------------------
[02:15:05] [Paper Watchdog Thread/ERROR]: Current Thread: Server thread
[02:15:05] [Paper Watchdog Thread/ERROR]: 	PID: 41 | Suspended: false | Native: false | State: RUNNABLE
[02:15:05] [Paper Watchdog Thread/ERROR]: 	Stack:
[02:15:05] [Paper Watchdog Thread/ERROR]: 		ca.spottedleaf.dataconverter.minecraft.MCDataConverter.convert(MCDataConverter.java:48)
[02:15:05] [Paper Watchdog Thread/ERROR]: 		ca.spottedleaf.dataconverter.minecraft.MCDataConverter.convertTag(MCDataConverter.java:29)
[02:15:05] [Paper Watchdog Thread/ERROR]: 		net.minecraft.world.level.chunk.storage.ChunkStorage.upgradeChunkTag(ChunkStorage.java:122)
[02:15:05] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.level.ChunkMap.upgradeChunkTag(ChunkMap.java:942)
[02:15:05] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.level.ChunkMap.readConvertChunkSync(ChunkMap.java:955)
[02:15:05] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.level.ChunkMap.readChunk(ChunkMap.java:933)
[02:15:05] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.level.ChunkMap.getChunkStatusOnDisk(ChunkMap.java:986)
[02:15:05] [Paper Watchdog Thread/ERROR]: 		org.bukkit.craftbukkit.v1_20_R1.CraftWorld.isChunkGenerated(CraftWorld.java:385)
[02:15:05] [Paper Watchdog Thread/ERROR]: 		org.bukkit.craftbukkit.v1_20_R1.CraftWorld.lambda$isChunkGenerated$2(CraftWorld.java:374)
[02:15:05] [Paper Watchdog Thread/ERROR]: 		org.bukkit.craftbukkit.v1_20_R1.CraftWorld$$Lambda$19998/0x0000000800e55a48.get(Unknown Source)
[02:15:05] [Paper Watchdog Thread/ERROR]: 		java.base@17.0.3/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768)
[02:15:05] [Paper Watchdog Thread/ERROR]: 		net.minecraft.util.thread.BlockableEventLoop.doRunTask(BlockableEventLoop.java:153)
[02:15:05] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.level.ServerChunkCache$MainThreadExecutor.doRunTask(ServerChunkCache.java:849)
[02:15:05] [Paper Watchdog Thread/ERROR]: 		net.minecraft.util.thread.BlockableEventLoop.pollTask(BlockableEventLoop.java:126)
[02:15:05] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.level.ServerChunkCache$MainThreadExecutor.pollTask(ServerChunkCache.java:858)
[02:15:05] [Paper Watchdog Thread/ERROR]: 		net.minecraft.util.thread.BlockableEventLoop.managedBlock(BlockableEventLoop.java:136)
[02:15:05] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.level.ServerChunkCache.getChunk(ServerChunkCache.java:300)
[02:15:05] [Paper Watchdog Thread/ERROR]: 		net.minecraft.world.level.Level.getChunk(Level.java:507)
[02:15:05] [Paper Watchdog Thread/ERROR]: 		net.minecraft.world.level.Level.getBlockState(Level.java:780)
[02:15:05] [Paper Watchdog Thread/ERROR]: 		net.minecraft.world.entity.Entity.getOnPos(Entity.java:1365)
[02:15:05] [Paper Watchdog Thread/ERROR]: 		net.minecraft.world.entity.Entity.getOnPosLegacy(Entity.java:1347)
[02:15:05] [Paper Watchdog Thread/ERROR]: 		net.minecraft.world.entity.Entity.getBlockStateOnLegacy(Entity.java:1917)
[02:15:05] [Paper Watchdog Thread/ERROR]: 		net.minecraft.world.entity.LivingEntity.onChangedBlock(LivingEntity.java:612)
[02:15:05] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.level.ServerPlayer.onChangedBlock(ServerPlayer.java:1520)
[02:15:05] [Paper Watchdog Thread/ERROR]: 		net.minecraft.world.entity.LivingEntity.baseTick(LivingEntity.java:466)
[02:15:05] [Paper Watchdog Thread/ERROR]: 		net.minecraft.world.entity.Entity.tick(Entity.java:787)
[02:15:05] [Paper Watchdog Thread/ERROR]: 		net.minecraft.world.entity.LivingEntity.tick(LivingEntity.java:2975)
[02:15:05] [Paper Watchdog Thread/ERROR]: 		net.minecraft.world.entity.player.Player.tick(Player.java:276)
[02:15:05] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.level.ServerPlayer.doTick(ServerPlayer.java:758)
[02:15:05] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.network.ServerGamePacketListenerImpl.tick(ServerGamePacketListenerImpl.java:359)
[02:15:05] [Paper Watchdog Thread/ERROR]: 		net.minecraft.network.Connection.tick(Connection.java:599)
[02:15:05] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.network.ServerConnectionListener.tick(ServerConnectionListener.java:234)
[02:15:05] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.MinecraftServer.tickChildren(MinecraftServer.java:1573)
[02:15:05] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.dedicated.DedicatedServer.tickChildren(DedicatedServer.java:447)
[02:15:05] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1394)
[02:15:05] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1171)
[02:15:05] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:318)
[02:15:05] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.MinecraftServer$$Lambda$4706/0x0000000801637dc0.run(Unknown Source)
[02:15:05] [Paper Watchdog Thread/ERROR]: 		java.base@17.0.3/java.lang.Thread.run(Thread.java:833)
[02:15:05] [Paper Watchdog Thread/ERROR]: ------------------------------
[02:15:05] [Paper Watchdog Thread/ERROR]: --- DO NOT REPORT THIS TO PAPER - THIS IS NOT A BUG OR A CRASH ---
[02:15:05] [Paper Watchdog Thread/ERROR]: ------------------------------
```

### Plugin and Datapack List

[02:18:25 INFO]: Server Plugins (145):
[02:18:25 INFO]: Bukkit Plugins:
[02:18:25 INFO]:  - AdvancedChests, AdvancedEnchantments, AdvancedVanish, *AfkWarner, ajLeaderboards, ajQueue, AlonsoTags, AntiVillagerLag, BeastCore, BeastWithdraw
[02:18:25 INFO]:  BlueSlimeCore, Bounties, BuildersWand, BuycraftX, ChatChat, ChatSentry, ChestSort, ChillTitles, ChunkHoppers, Chunky
[02:18:25 INFO]:  ChunkyBorder, Citizens, ClaimChat, CMILib, CoinCurrency, CombatLogX, CooldownsX, CoreProtect, CosmicItems, *CosmicPlaytime
[02:18:25 INFO]:  CosmosChallenges, CosmosCore, CosmosLogin, CosmosStacker, CrazyCrates, CustomChatTags, CustomFishing, DecentHolograms, DeluxeMenus, DeluxeSellwands
[02:18:25 INFO]:  DiscordCommands, DiscordSRV, DragonEggDrop, DuelAnywhere, eGlow, Elevator, EpicRename, Essentials, EssentialsSpawn, Exchanger
[02:18:25 INFO]:  ExecutableItems, EZColors, FartherViewDistance, FastAsyncWorldEdit, FastSpawners, floodgate, GeyserTabFix, GPClaimExpiration, GriefPrevention, GriefPreventionFlags
[02:18:25 INFO]:  GriefPreventionTP, GSit, HeadDatabase, helper, helper-profiles, helper-sql, HiddenArmor, *HideItem, HomeGUI, ImageMaps
[02:18:25 INFO]:  Infiniteannouncements, InfiniteBlocks, InventoryRollbackPlus, InvisibleItemFrames, ItemsAdder, ItemSkins, JakesRTP, *JoinLeave, KixsChatGames, LastLoginAPI
[02:18:25 INFO]:  LibsDisguises, LiteBans, LockettePro, LoneLibs, LuckPerms, MCMetrics, mcMMO, MobFarmManager, Multiverse-Core, MyCommand
[02:18:25 INFO]:  NBTAPI, NoClone, NotTooExpensive, *OfflineCommandFix, OpenInv, ParticleClaims, Parties, PartyAndFriendsGUI, PinataParty, Pl-Hide-Pro
[02:18:25 INFO]:  Pl3xmap-GriefPrevention, PlaceholderAPI, PlayerVaults, PlayerWarps, PlugManX, ProtocolLib, PyroFishingPro, PyroLib, Quests, QuickShop
[02:18:25 INFO]:  RealEstate, SafariNet, SchedulerPlugin, SCore, Scyther, ShopGUIPlus, SilkSpawners, spark, SpawnerCollectors, squaremap
[02:18:25 INFO]:  StaffChat, StatTrak, SuperPouches, SurvivalCore, TAB, TradeMe, TradeMeAddon, UltimateAutoRestart, Vault, Vehicles
[02:18:25 INFO]:  VehiclesLimiter, VeinMiner, ViaBackwards, ViaVersion, VillagerManager, *VoidGenerator, Votifier, VotingPlugin, Vulcan, WildLoaders
[02:18:25 INFO]:  WorldGuard, WorldGuardExtraFlags, zAuctionHouse-Discord, zAuctionHouseV3, zKoth

### Actions to reproduce (if known)

have about 100 players online. This is occurring on all of my gamemodes, completely separate from eachother.

### Paper version

[02:19:04 INFO]: Checking version, please wait...
[02:19:05 INFO]: This server is running Paper version git-Paper-36 (MC: 1.20.1) (Implementing API version 1.20.1-R0.1-SNAPSHOT) (Git: faf9a65)

**Comment by spookymgmt** [comment]
My resource world, where the error came from above, was created ON 1.20, and I genned the entire thing.

**Comment by spookymgmt** [comment]
I've also received similar error while on pufferfish, all though I am now currently on paper.


```[22:39:34] [Paper Watchdog Thread/ERROR]: --- DO NOT REPORT THIS TO PAPER - THIS IS NOT A BUG OR A CRASH  - git-Pufferfish-4 (MC: 1.20.1) ---
[22:39:34] [Paper Watchdog Thread/ERROR]: The server has not responded for 10 seconds! Creating thread dump
[22:39:34] [Paper Watchdog Thread/ERROR]: ------------------------------
[22:39:34] [Paper Watchdog Thread/ERROR]: Server thread dump (Look for plugins here before reporting to Paper!):
[22:39:34] [Paper Watchdog Thread/ERROR]: [ChunkTaskScheduler] Chunk wait task info below: 
[22:39:34] [Paper Watchdog Thread/ERROR]: [ChunkTaskScheduler] Chunk wait: [( -793,-11) in 'resourceworld']
[22:39:34] [Paper Watchdog Thread/ERROR]: [ChunkTaskScheduler] Chunk holder: NewChunkHolder{world=resourceworld, chunkX=-793, chunkZ=-11, entityChunkFromDisk=false, lastChunkCompletion={chunk_class=net.minecraft.world.level.chunk.ProtoChunk,status=minecraft:surface}, currentGenStatus=minecraft:surface, requestedGenStatus=minecraft:full, generationTask=ChunkProgressionTask{class: io.papermc.paper.chunk.system.scheduling.ChunkUpgradeGenericStatusTask, for world: resourceworld, chunk: (-793,-11), hashcode: 829559603, priority: COMPLETING, status: minecraft:carvers, scheduled: true}, generationTaskStatus=minecraft:carvers, priority=BLOCKING, priorityLocked=false, neighbourRequestedPriority=IDLE, effective_priority=BLOCKING, oldTicketLevel=33, currentTicketLevel=33, totalNeighboursUsingThisChunk=1, fullNeighbourChunksLoadedBitset=0, chunkStatusRaw=0, currentChunkStatus=INACCESSIBLE, pendingChunkStatus=INACCESSIBLE, is_unload_safe=ticket_level, killed=false}
[22:39:34] [Paper Watchdog Thread/ERROR]: Ticking entity: minecraft:player, entity class: net.minecraft.server.level.EntityPlayer
[22:39:34] [Paper Watchdog Thread/ERROR]: Entity status: removed: false, valid: true, alive: true, is passenger: false
[22:39:34] [Paper Watchdog Thread/ERROR]: Entity UUID: 1b453600-a345-4b1e-8bcf-a8489ec3c8c5
[22:39:34] [Paper Watchdog Thread/ERROR]: Position: world: 'resourceworld' at location (1920.5, 116.0, -3084.5)
[22:39:34] [Paper Watchdog Thread/ERROR]: Velocity: (0.0, 0.0, 0.0) (in blocks per tick)
[22:39:34] [Paper Watchdog Thread/ERROR]: Entity AABB: AABB[1920.199999988079, 116.0, -3084.800000011921] -> [1920.800000011921, 117.79999995231628, -3084.199999988079]
[22:39:34] [Paper Watchdog Thread/ERROR]: Total packets processed on the main thread for all players: 59897032
[22:39:34] [Paper Watchdog Thread/ERROR]: ------------------------------
[22:39:34] [Paper Watchdog Thread/ERROR]: Current Thread: Server thread
[22:39:34] [Paper Watchdog Thread/ERROR]: 	PID: 41 | Suspended: false | Native: false | State: TIMED_WAITING
[22:39:34] [Paper Watchdog Thread/ERROR]: 	Stack:
[22:39:34] [Paper Watchdog Thread/ERROR]: 		java.base@17.0.3/jdk.internal.misc.Unsafe.park(Native Method)
[22:39:34] [Paper Watchdog Thread/ERROR]: 		java.base@17.0.3/java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:252)
[22:39:34] [Paper Watchdog Thread/ERROR]: 		net.minecraft.util.thread.BlockableEventLoop.waitForTasks(BlockableEventLoop.java:148)
[22:39:34] [Paper Watchdog Thread/ERROR]: 		net.minecraft.util.thread.BlockableEventLoop.managedBlock(BlockableEventLoop.java:137)
[22:39:34] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.level.ServerChunkCache.getChunk(ServerChunkCache.java:303)
[22:39:34] [Paper Watchdog Thread/ERROR]: 		net.minecraft.world.level.Level.getChunk(Level.java:610)
[22:39:34] [Paper Watchdog Thread/ERROR]: 		net.minecraft.world.level.Level.getBlockState(Level.java:883)
[22:39:34] [Paper Watchdog Thread/ERROR]: 		net.minecraft.world.entity.Entity.getOnPos(Entity.java:1399)
[22:39:34] [Paper Watchdog Thread/ERROR]: 		net.minecraft.world.entity.Entity.getOnPosLegacy(Entity.java:1381)
[22:39:34] [Paper Watchdog Thread/ERROR]: 		net.minecraft.world.entity.Entity.getBlockStateOnLegacy(Entity.java:1951)
[22:39:34] [Paper Watchdog Thread/ERROR]: 		net.minecraft.world.entity.LivingEntity.onChangedBlock(LivingEntity.java:611)
[22:39:34] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.level.ServerPlayer.onChangedBlock(ServerPlayer.java:1520)
[22:39:34] [Paper Watchdog Thread/ERROR]: 		net.minecraft.world.entity.LivingEntity.baseTick(LivingEntity.java:465)
[22:39:34] [Paper Watchdog Thread/ERROR]: 		net.minecraft.world.entity.Entity.tick(Entity.java:815)
[22:39:34] [Paper Watchdog Thread/ERROR]: 		net.minecraft.world.entity.LivingEntity.tick(LivingEntity.java:2997)
[22:39:34] [Paper Watchdog Thread/ERROR]: 		net.minecraft.world.entity.player.Player.tick(Player.java:276)
[22:39:34] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.level.ServerPlayer.doTick(ServerPlayer.java:758)
[22:39:34] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.network.ServerGamePacketListenerImpl.tick(ServerGamePacketListenerImpl.java:359)
[22:39:34] [Paper Watchdog Thread/ERROR]: 		net.minecraft.network.Connection.tick(Connection.java:599)
[22:39:34] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.network.ServerConnectionListener.tick(ServerConnectionListener.java:234)
[22:39:34] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.MinecraftServer.tickChildren(MinecraftServer.java:1575)
[22:39:34] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.dedicated.DedicatedServer.tickChildren(DedicatedServer.java:450)
[22:39:34] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1396)
[22:39:34] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1173)
[22:39:34] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:320)
[22:39:34] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.MinecraftServer$$Lambda$4715/0x0000000801635c40.run(Unknown Source)
[22:39:34] [Paper Watchdog Thread/ERROR]: 		java.base@17.0.3/java.lang.Thread.run(Thread.java:833)
[22:39:34] [Paper Watchdog Thread/ERROR]: ------------------------------
[22:39:34] [Paper Watchdog Thread/ERROR]: --- DO NOT REPORT THIS TO PAPER - THIS IS NOT A BUG OR A CRASH ---
[22:39:34] [Paper Watchdog Thread/ERROR]: ------------------------------```

**Comment by spookymgmt** [comment]
```[22:39:39] [Paper Watchdog Thread/ERROR]: --- DO NOT REPORT THIS TO PAPER - THIS IS NOT A BUG OR A CRASH  - git-Pufferfish-4 (MC: 1.20.1) ---
[22:39:39] [Paper Watchdog Thread/ERROR]: The server has not responded for 15 seconds! Creating thread dump
[22:39:39] [Paper Watchdog Thread/ERROR]: ------------------------------
[22:39:39] [Paper Watchdog Thread/ERROR]: Server thread dump (Look for plugins here before reporting to Paper!):
[22:39:39] [Paper Watchdog Thread/ERROR]: [ChunkTaskScheduler] Chunk wait task info below: 
[22:39:39] [Paper Watchdog Thread/ERROR]: [ChunkTaskScheduler] Chunk wait: [( -793,-11) in 'resourceworld']
[22:39:39] [Paper Watchdog Thread/ERROR]: [ChunkTaskScheduler] Chunk holder: NewChunkHolder{world=resourceworld, chunkX=-793, chunkZ=-11, entityChunkFromDisk=false, lastChunkCompletion={chunk_class=net.minecraft.world.level.chunk.ProtoChunk,status=minecraft:carvers}, currentGenStatus=minecraft:carvers, requestedGenStatus=minecraft:full, generationTask=null, generationTaskStatus=null, priority=BLOCKING, priorityLocked=false, neighbourRequestedPriority=IDLE, effective_priority=BLOCKING, oldTicketLevel=33, currentTicketLevel=33, totalNeighboursUsingThisChunk=8, fullNeighbourChunksLoadedBitset=0, chunkStatusRaw=0, currentChunkStatus=INACCESSIBLE, pendingChunkStatus=INACCESSIBLE, is_unload_safe=ticket_level, killed=false}
[22:39:39] [Paper Watchdog Thread/ERROR]: Ticking entity: minecraft:player, entity class: net.minecraft.server.level.EntityPlayer
[22:39:39] [Paper Watchdog Thread/ERROR]: Entity status: removed: false, valid: true, alive: true, is passenger: false
[22:39:39] [Paper Watchdog Thread/ERROR]: Entity UUID: 1b453600-a345-4b1e-8bcf-a8489ec3c8c5
[22:39:39] [Paper Watchdog Thread/ERROR]: Position: world: 'resourceworld' at location (1920.5, 116.0, -3084.5)
[22:39:39] [Paper Watchdog Thread/ERROR]: Velocity: (0.0, 0.0, 0.0) (in blocks per tick)
[22:39:39] [Paper Watchdog Thread/ERROR]: Entity AABB: AABB[1920.199999988079, 116.0, -3084.800000011921] -> [1920.800000011921, 117.79999995231628, -3084.199999988079]
[22:39:39] [Paper Watchdog Thread/ERROR]: Total packets processed on the main thread for all players: 59897032
[22:39:39] [Paper Watchdog Thread/ERROR]: ------------------------------
[22:39:39] [Paper Watchdog Thread/ERROR]: Current Thread: Server thread
[22:39:39] [Paper Watchdog Thread/ERROR]: 	PID: 41 | Suspended: false | Native: false | State: TIMED_WAITING
[22:39:39] [Paper Watchdog Thread/ERROR]: 	Stack:
[22:39:39] [Paper Watchdog Thread/ERROR]: 		java.base@17.0.3/jdk.internal.misc.Unsafe.park(Native Method)
[22:39:39] [Paper Watchdog Thread/ERROR]: 		java.base@17.0.3/java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:252)
[22:39:39] [Paper Watchdog Thread/ERROR]: 		net.minecraft.util.thread.BlockableEventLoop.waitForTasks(BlockableEventLoop.java:148)
[22:39:39] [Paper Watchdog Thread/ERROR]: 		net.minecraft.util.thread.BlockableEventLoop.managedBlock(BlockableEventLoop.java:137)
[22:39:39] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.level.ServerChunkCache.getChunk(ServerChunkCache.java:303)
[22:39:39] [Paper Watchdog Thread/ERROR]: 		net.minecraft.world.level.Level.getChunk(Level.java:610)
[22:39:39] [Paper Watchdog Thread/ERROR]: 		net.minecraft.world.level.Level.getBlockState(Level.java:883)
[22:39:39] [Paper Watchdog Thread/ERROR]: 		net.minecraft.world.entity.Entity.getOnPos(Entity.java:1399)
[22:39:39] [Paper Watchdog Thread/ERROR]: 		net.minecraft.world.entity.Entity.getOnPosLegacy(Entity.java:1381)
[22:39:39] [Paper Watchdog Thread/ERROR]: 		net.minecraft.world.entity.Entity.getBlockStateOnLegacy(Entity.java:1951)
[22:39:39] [Paper Watchdog Thread/ERROR]: 		net.minecraft.world.entity.LivingEntity.onChangedBlock(LivingEntity.java:611)
[22:39:39] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.level.ServerPlayer.onChangedBlock(ServerPlayer.java:1520)
[22:39:39] [Paper Watchdog Thread/ERROR]: 		net.minecraft.world.entity.LivingEntity.baseTick(LivingEntity.java:465)
[22:39:39] [Paper Watchdog Thread/ERROR]: 		net.minecraft.world.entity.Entity.tick(Entity.java:815)
[22:39:39] [Paper Watchdog Thread/ERROR]: 		net.minecraft.world.entity.LivingEntity.tick(LivingEntity.java:2997)
[22:39:39] [Paper Watchdog Thread/ERROR]: 		net.minecraft.world.entity.player.Player.tick(Player.java:276)
[22:39:39] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.level.ServerPlayer.doTick(ServerPlayer.java:758)
[22:39:39] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.network.ServerGamePacketListenerImpl.tick(ServerGamePacketListenerImpl.java:359)
[22:39:39] [Paper Watchdog Thread/ERROR]: 		net.minecraft.network.Connection.tick(Connection.java:599)
[22:39:39] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.network.ServerConnectionListener.tick(ServerConnectionListener.java:234)
[22:39:39] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.MinecraftServer.tickChildren(MinecraftServer.java:1575)
[22:39:39] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.dedicated.DedicatedServer.tickChildren(DedicatedServer.java:450)
[22:39:39] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1396)
[22:39:39] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1173)
[22:39:39] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:320)
[22:39:39] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.MinecraftServer$$Lambda$4715/0x0000000801635c40.run(Unknown Source)
[22:39:39] [Paper Watchdog Thread/ERROR]: 		java.base@17.0.3/java.lang.Thread.run(Thread.java:833)
[22:39:39] [Paper Watchdog Thread/ERROR]: ------------------------------
[22:39:39] [Paper Watchdog Thread/ERROR]: --- DO NOT REPORT THIS TO PAPER - THIS IS NOT A BUG OR A CRASH ---
[22:39:39] [Paper Watchdog Thread/ERROR]: ------------------------------```

**Comment by Redned235** [comment]
Your server is lagging. Spark is probably your best bet for figuring out the root of performance issues - very unlikely a Paper issue.

Also from the log:
> DO NOT REPORT THIS TO PAPER - THIS IS NOT A BUG OR A CRASH ---

**Comment by spookymgmt** [comment]
> 

yeah no, this is a bug. My server is lagging due to a bug. Spikes causing very very long ms ticks.

**Comment by spookymgmt** [comment]
https://spark.lucko.me/pJPF7psj0w is 1.20 worse than 1.13 now?

**Comment by electronicboy** [comment]
Something induced a sync chunk load, would need an all threads spark report to see what's going on properly

**Comment by spookymgmt** [comment]
https://spark.lucko.me/Zg4SGeyeZO

https://spark.lucko.me/1lrsMHARTJ

The above sparks were ran while the error WAS NOT happening, but tps was generally 10x worse than usual.


I will also send a spark when the error IS happening later today.

**Comment by electronicboy** [comment]
one reprort is pretty heavily sunk by somebody having a bunch of itemframes with maps from the looks of it; The other is the 11k entities you have on the server

**Comment by Redned235** [comment]
You may have some luck using the Paperclip version from this PR: https://github.com/PaperMC/Paper/pull/9377. It includes a patch that optimizes maps on item frames if nobody can see them. It may help a bit in your case, but don't expect it to be a silver bullet.

**Comment by Boy0000** [comment]
> It includes a patch that optimizes maps on item frames if nobody can see them

Is there a reason it only covers ItemFrames? Know the main issue of the report was ItemFrames but figured this could be the case for other entities aswell, like Display Entities?

**Comment by Redned235** [comment]
> Is there a reason it only covers ItemFrames? Know the main issue of the report was ItemFrames but figured this could be the case for other entities aswell, like Display Entities?

There's a lot of additional tracking logic that occurs inside of the ServerEntity class which needs to be updated, so simply just ignoring all this logic causes a few desyncs (that is originally what the PR did, and issues were found with that approach). Additionally, display entities are often used in far smaller quantities than item frames, which are available in survival mode, so any benefit here is marginal at best.

**Comment by spookymgmt** [comment]
Backing up, then updating now. Will update with new sparks during peak times in around 3-6 hours.

**Comment by spookymgmt** [comment]
Using the new jar, here is a spark for TWO DIFFERENT servers, checking for ticks over 350

https://spark.lucko.me/SlzDNZ8wGP
https://spark.lucko.me/8ChZnvwbjh


Heres a **thread * ** spark, each spark is from different servers/different gamemodes, while a very long MS tick occurred.

https://spark.lucko.me/4govqwticS
https://spark.lucko.me/ipbmuJ9VNu
https://spark.lucko.me/JRdX624krn

**Comment by unclesom** [comment]
@spookymgmt This is offtopic and I have no way of contacting you but my network is currently having the same issue that you had in January where one player with packet issues is making the server un-joinable. Did you end up fixing the problem? I'm desperate in finding the solution so I apologize for posting an offtopic comment here

**Comment by lynxplay** [comment]
Hopefully resolved by dadd8b5a7cf3775f9d9778f17bf7a0bedfe0a768. Please comment on here for us to reopen this if the newest build still show this issue.
